dashboard_server.py

```
@@ -0,0 +1,530 @@
+                if existing and existing.get('timestamp') and existing['timestamp'] >= detailed_ts:
+                    # Existing is newer or equal; skip older run
+                    continue
+                results[model_name] = {
+                    'detailed': df.to_dict('records'),
+                    'summary': summary_data,
+                    'csv_file': csv_file,
+                    'summary_file': selected_summary,
+                    'timestamp': detailed_ts,
+                }
+                
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)